src/object_store/src/object/s3.rs

```diff
@@ -675,25 +678,26 @@ impl S3ObjectStore {
     fn obj_store_request(
         &self,
         path: &str,
-        start_pos: Option<usize>,
-        end_pos: Option<usize>,
+        range: impl RangeBounds<usize> + Clone + Send + Sync + 'static,
     ) -> GetObjectFluentBuilder {
         let req = self.client.get_object().bucket(&self.bucket).key(path);
 
-        match (start_pos, end_pos) {
-            (None, None) => {
-                // No range is given. Return request as is.
-                req
-            }
-            _ => {
-                // At least one boundary is given. Return request with range limitation.
-                req.range(format!(
-                    "bytes={}-{}",
-                    start_pos.map_or(String::new(), |pos| pos.to_string()),
-                    end_pos.map_or(String::new(), |pos| pos.to_string())
-                ))
-            }
+        if range.start_bound() == Bound::Unbounded && range.end_bound() == Bound::Unbounded {
+            return req;
         }
+
+        let start = match range.start_bound() {
```

Comment:
How about constructing start and end in separate functions? It's repeated several times in pr

Reply:
Agreed! I'm doing it.